### PR TITLE
Feature/user login

### PR DESCRIPTION
**Pull request by Armengedon**
Login implementation using spring security
tested with postman -> Basic security login

@ortizdaniel @Arnau95 @Bothar

### CODE REVIEW

**Review by ortizdaniel — APPROVED** [review]

File: src/main/kotlin/com/example/demo/DemoApplication.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.example.demo
 
 import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.context.properties.EnableConfigurationProperties
```
Comment (macmagic) [inline src/main/kotlin/com/example/demo/DemoApplication.kt:4]: Import not used here

File: src/main/kotlin/com/example/demo/controllers/UserController.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@RestController
+@RequestMapping("/user")
+class UserController(private val repository: UserRepository) {
+
+    @GetMapping
+    fun findAll() = repository.findAll()
+
+    @GetMapping("/{login}")
```
Comment (macmagic) [inline src/main/kotlin/com/example/demo/controllers/UserController.kt:25]: Bad practice, path variable is used to handle variables in the URI part.

Change the controller RequestMapping to /user/login and map GetMapping and PostMapping directly

File: src/main/kotlin/com/example/demo/controllers/UserController.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    fun findOne(@PathVariable login: String): User {
+        return repository.findByLogin(login) ?: throw ResponseStatusException(
```
Comment (macmagic) [inline src/main/kotlin/com/example/demo/controllers/UserController.kt:27]: Controllers never interact directly with repositories, use Service to add the logical business

File: src/main/kotlin/com/example/demo/repositories/UserRepository.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.example.demo.repositories
+
+import com.example.demo.entities.User
+import org.springframework.data.repository.CrudRepository
+
+interface UserRepository : CrudRepository<User, Long> {
```
Comment (macmagic) [inline src/main/kotlin/com/example/demo/repositories/UserRepository.kt:6]: Why don't use JPA library?

File: src/test/kotlin/com/example/demo/RepositoriesTests.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    val entityManager: TestEntityManager,
+    val userRepository: UserRepository
+) {
+
+    @BeforeAll
+    fun prepare() {
+    }
+
+    @Test
+    fun `When findByLogin then return User`() {
```
Comment (macmagic) [inline src/test/kotlin/com/example/demo/RepositoriesTests.kt:24]: Is not a relevant test, why do you test the third-party library? The library has its own tests to check the internal functionality

**Pull request merged**